File: order-service/src/main/java/com/paulopacifico/orderservice/messaging/api/OrderShipmentFailedEvent.java
package com.paulopacifico.orderservice.messaging.api;

import java.time.OffsetDateTime;
import java.util.UUID;

public record OrderShipmentFailedEvent(
        UUID eventId,
        Long orderId,
        String orderNumber,
        String reason,
        OffsetDateTime occurredAt
) {
}

File: inventory-service/src/main/kotlin/com/paulopacifico/inventoryservice/messaging/api/OrderShipmentFailedEvent.kt
package com.paulopacifico.inventoryservice.messaging.api

import java.time.OffsetDateTime
import java.util.UUID

data class OrderShipmentFailedEvent(
    val eventId: UUID,
    val orderId: Long,
    val orderNumber: String,
    val reason: String,
    val occurredAt: OffsetDateTime,
)
